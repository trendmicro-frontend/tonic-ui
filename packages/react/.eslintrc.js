module.exports = {
  extends: 'trendmicro',
  parser: '@babel/eslint-parser',
  env: {
    browser: true,
    node: true,
    jest: true,
  },
  plugins: [
    '@babel',
    'react-hooks',
  ],
  rules: {
    'react/prop-types': 0,
    'react-hooks/rules-of-hooks': 'error', // Checks rules of Hooks
    'react-hooks/exhaustive-deps': 'error', // Checks effect dependencies
  },
};
