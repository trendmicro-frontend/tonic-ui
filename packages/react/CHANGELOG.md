# @tonic-ui/react

## 2.0.3

### Patch Changes

- fix(react/OverflowTooltip): resolve tooltip misalignment issue with popup menu items by [@tinaClin](https://github.com/tinaClin) in [#894](https://github.com/trendmicro-frontend/tonic-ui/pull/894)

- feat(react): improve tooltip placement and offset calculations by [@cheton](https://github.com/cheton) in [#897](https://github.com/trendmicro-frontend/tonic-ui/pull/897)
