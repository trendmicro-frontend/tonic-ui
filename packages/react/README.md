React UI components built with Emotion and Styled System
