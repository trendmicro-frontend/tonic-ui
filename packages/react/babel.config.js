module.exports = {
  extends: '@trendmicro/babel-config',
  presets: [
    '@babel/preset-env',
    ['@babel/preset-react', {
      /**
       * Decides which runtime to use.
       * - 'automatic' auto imports the functions that JSX transpiles to.
       * - 'classic' does not automatic import anything.
       */
      'runtime': 'classic', // defaults to classic
      //'runtime': 'automatic', // for React 16.14.0, React 17 and higher that supports the new transform
    }],
    '@emotion/babel-preset-css-prop',
  ],
};
