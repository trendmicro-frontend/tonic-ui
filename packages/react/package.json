{
  "name": "@tonic-ui/react",
  "version": "1.0.0-alpha",
  "description": "React Tonic UI component library.",
  "main": "dist/index.js",
  "module": "dist/es/index.js",
  "publishConfig": {
    "access": "public",
    "directory": "build"
  },
  "scripts": {
    "build": "cross-env PACKAGE_NAME=index INPUT=src OUTPUT_DIRECTORY=dist rollup --config rollup.config.js",
    "clean": "del build coverage dist",
    "eslint": "eslint --ext .js,.jsx,.mjs .",
    "postbuild": "cpy --parents {dist/**,LICENSE,README.md,.npmignore,package.json} build",
    "pre-push": "yarn run eslint",
    "prepublish": "yarn run build",
    "test": "jest"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "1.x",
    "@emotion/react": "11.x",
    "@emotion/styled": "11.x",
    "@popperjs/core": "2.x",
    "@tonic-ui/react-hooks": "^1.0.0-alpha",
    "@tonic-ui/styled-system": "^1.0.0-alpha",
    "@tonic-ui/theme": "^1.0.0-alpha",
    "@trendmicro/tmicon": "^1.21.0",
    "chained-function": "^0.5.0",
    "color": "3.x",
    "ensure-type": "1.x",
    "lodash.get": "4.x",
    "micro-memoize": "4.x",
    "prop-types": "15.x",
    "react-focus-lock": "2.x",
    "react-is": "^16.8.0 || ^17",
    "react-transition-group": "4.x"
  },
  "devDependencies": {
    "@babel/cli": "^7.0.0",
    "@babel/core": "^7.0.0",
    "@babel/eslint-parser": "^7.0.0",
    "@babel/eslint-plugin": "^7.0.0",
    "@babel/plugin-transform-runtime": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@babel/preset-react": "^7.0.0",
    "@emotion/babel-preset-css-prop": "~11.2.0",
    "@emotion/jest": "~11.5.0",
    "@rollup/plugin-node-resolve": "~11.2.0",
    "@testing-library/jest-dom": "^5.16.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/react-hooks": "^7.0.2",
    "@testing-library/user-event": "^13.5.0",
    "@trendmicro/babel-config": "^1.0.2",
    "babel-loader": "~8.2.3",
    "cpy-cli": "~3.1.1",
    "cross-env": "~7.0.3",
    "del-cli": "~4.0.1",
    "eslint": "^7.0.0",
    "eslint-config-trendmicro": "^2.1.0",
    "eslint-loader": "~4.0.2",
    "eslint-plugin-import": "latest",
    "eslint-plugin-jsx-a11y": "latest",
    "eslint-plugin-react": "latest",
    "eslint-plugin-react-hooks": "latest",
    "jest": "~27.3.1",
    "jest-extended": "^1.1.0",
    "react": "^16.8.0 || ^17",
    "react-dom": "^16.8.0 || ^17",
    "react-test-renderer": "^16.8.0 || ^17",
    "rollup": "~2.42.3",
    "rollup-plugin-babel": "~4.4.0"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17",
    "react-dom": "^16.8.0 || ^17"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.js"
    ],
    "coverageReporters": [
      "lcov",
      "text",
      "html"
    ],
    "setupFiles": [],
    "setupFilesAfterEnv": [
      "<rootDir>/jest-setup.js"
    ],
    "testEnvironment": "jsdom",
    "testMatch": [
      "<rootDir>/src/**/*.test.js",
      "<rootDir>/test/**/*.js"
    ]
  },
  "author": "Cheton Wu <cheton_wu@trendmicro.com>",
  "homepage": "https://github.com/trendmicro-frontend/tonic-ui",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/trendmicro-frontend/tonic-ui.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/trendmicro-frontend/tonic-ui/issues"
  },
  "keywords": [
    "components",
    "emotion",
    "react",
    "styled-system",
    "tonic-ui"
  ]
}
