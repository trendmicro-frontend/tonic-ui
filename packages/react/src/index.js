// components
export * from './accordion';
export * from './alert';
export * from './badge';
export * from './box';
export * from './button';
export * from './checkbox';
export * from './color-mode';
export * from './color-style';
export * from './css-baseline';
export * from './divider';
export * from './drawer';
export * from './flat-button';
export * from './flex';
export * from './grid';
export * from './icon';
export * from './image';
export * from './input';
export * from './link';
export * from './menu';
export * from './modal';
export * from './pagination';
export * from './popover';
export * from './popper';
export * from './portal';
export * from './presence';
export * from './radio';
export * from './search-input';
export * from './scrollbar';
export * from './select';
export * from './skeleton';
export * from './space';
export * from './spinner';
export * from './stack';
export * from './switch';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text';
export * from './textarea';
export * from './theme';
export * from './toast';
export * from './tooltip';
export * from './transitions';
export * from './visually-hidden';

// hooks
export useDisclosure from './deprecated/useDisclosure'; // deprecated

// settings
export colorStyle from './shared/color-style';
export theme from './shared/theme';
