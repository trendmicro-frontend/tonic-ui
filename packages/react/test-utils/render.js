import { matchers } from '@emotion/jest';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import * as React from 'react';
import {
  PortalManager,
  ToastManager,
  TonicProvider,
  theme,
} from '../src';

// https://emotion.sh/docs/@emotion/jest#tohavestylerule
//
// Add the custom matchers provided by '@emotion/jest'
expect.extend(matchers);

const customRender = (ui, options) => {
  const wrapper = ({ children }) => (
    <TonicProvider
      colorMode={{
        defaultValue: 'dark',
      }}
      theme={{
        ...theme,
        config: {
          ...theme.config,
          useCSSVariables: options?.useCSSVariables,
        },
      }}
    >
      <PortalManager>
        <ToastManager>
          {children}
        </ToastManager>
      </PortalManager>
    </TonicProvider>
  );

  return render(ui, { wrapper, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
