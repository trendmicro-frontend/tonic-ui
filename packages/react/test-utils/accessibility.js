import { act } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import * as React from 'react';
import { render } from './render';

expect.extend(toHaveNoViolations);

const testA11y = async (ui, options) => {
  const { axeOptions, ...rest } = { ...options };
  const container = React.isValidElement(ui)
    ? render(ui, rest).container
    : ui;
  await act(async () => {
    const results = await axe(container, axeOptions);
    expect(results).toHaveNoViolations();
  });
};

export {
  testA11y,
};
