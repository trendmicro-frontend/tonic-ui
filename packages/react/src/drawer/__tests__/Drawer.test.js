import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { render } from '@tonic-ui/react/test-utils/render';
import { testA11y } from '@tonic-ui/react/test-utils/accessibility';
import {
  Button,
  Input,
  Drawer,
  DrawerOverlay,
  DrawerContent,
  DrawerHeader,
  DrawerBody,
  DrawerFooter,
  DrawerCloseButton,
  Text,
} from '@tonic-ui/react/src';
import React, { useCallback, useRef, useState } from 'react';

describe('Drawer', () => {
  it('should render correctly', async () => {
    const renderOptions = {
      useCSSVariables: true,
    };
    const { baseElement } = render((
      <Drawer
        isOpen
        onClose={jest.fn()}
        placement="right"
      >
        <DrawerOverlay />
        <DrawerContent>
          <DrawerHeader>Drawer header</DrawerHeader>
          <DrawerBody>Drawer body</DrawerBody>
          <DrawerFooter>Drawer footer</DrawerFooter>
          <DrawerCloseButton />
        </DrawerContent>
      </Drawer>
    ), renderOptions);

    expect(baseElement).toMatchSnapshot();

    // Test baseElement because the drawer is rendered in a portal
    await testA11y(baseElement, {
      axeOptions: {
        rules: {
          // ARIA dialog and alertdialog nodes should have an accessible name (aria-dialog-name)
          'aria-dialog-name': { enabled: false },
        },
      },
    });
  });

  it('should have the proper ARIA attributes', () => {
    render(
      <Drawer
        isOpen
        onClose={jest.fn()}
        placement="right"
      >
        <DrawerOverlay />
        <DrawerContent data-testid="drawer-content">
          <DrawerHeader>Drawer header</DrawerHeader>
          <DrawerBody>Drawer body</DrawerBody>
          <DrawerFooter>Drawer footer</DrawerFooter>
          <DrawerCloseButton data-testid="drawer-close-button" />
        </DrawerContent>
      </Drawer>
    );

    const drawerContent = screen.getByTestId('drawer-content');
    const drawerCloseButton = screen.getByTestId('drawer-close-button');
    expect(drawerContent).toHaveAttribute('aria-modal', 'true');
    expect(drawerContent).toHaveAttribute('role', 'dialog');
    expect(drawerCloseButton).toHaveAttribute('aria-label', 'Close');
  });

  it('should fire `onClose` when the close button is clicked', async () => {
    const user = userEvent.setup();
    const onClose = jest.fn();

    render(
      <Drawer
        isOpen
        onClose={onClose}
        placement="right"
      >
        <DrawerOverlay />
        <DrawerContent>
          <DrawerHeader>Drawer header</DrawerHeader>
          <DrawerCloseButton data-testid="drawer-close-button" />
        </DrawerContent>
      </Drawer>
    );

    await user.click(screen.getByTestId('drawer-close-button'));
    expect(onClose).toHaveBeenCalled();
  });

  it('should close when pressing the escape key', async () => {
    const user = userEvent.setup();
    const onClose = jest.fn();

    render(
      <Drawer
        closeOnEsc
        ensureFocus
        isOpen
        onClose={onClose}
        placement="right"
      >
        <DrawerOverlay />
        <DrawerContent>
          <DrawerHeader>Drawer header</DrawerHeader>
          <DrawerBody>Drawer body</DrawerBody>
        </DrawerContent>
      </Drawer>
    );

    await user.keyboard('[Escape]');
    expect(onClose).toHaveBeenCalled();
  });

  it('should close when clicking outside the drawer', async () => {
    const user = userEvent.setup();
    const onClose = jest.fn();

    render(
      <Drawer
        closeOnOutsideClick
        isOpen
        onClose={onClose}
        placement="right"
      >
        <DrawerOverlay />
        <DrawerContent>
          <DrawerHeader>Drawer header</DrawerHeader>
          <DrawerBody>Drawer body</DrawerBody>
        </DrawerContent>
      </Drawer>
    );

    const dialog = await screen.findByRole('dialog');
    await user.click(dialog.parentElement);
    expect(onClose).toHaveBeenCalled();
  });

  it('should ensure proper focus management when opening and closing the drawer', async () => {
    const user = userEvent.setup();

    const TestComponent = () => {
      const [isOpen, setIsOpen] = useState(false);
      const openDrawer = useCallback(() => setIsOpen(true), []);
      const closeDrawer = useCallback(() => setIsOpen(false), []);
      const buttonRef = useRef(null);
      const inputRef = useRef(null);

      return (
        <>
          <Button
            data-testid="button"
            onClick={openDrawer}
            ref={buttonRef}
          >
            Open
          </Button>
          <Drawer
            ensureFocus
            finalFocusRef={buttonRef}
            initialFocusRef={inputRef}
            isOpen={isOpen}
            onClose={closeDrawer}
            placement="right"
          >
            <DrawerOverlay />
            <DrawerContent>
              <DrawerHeader>Drawer header</DrawerHeader>
              <DrawerBody>
                <Text>Drawer body</Text>
                <Input data-testid="input" ref={inputRef} />
              </DrawerBody>
              <DrawerCloseButton data-testid="drawer-close-button" />
            </DrawerContent>
          </Drawer>
        </>
      );
    };

    render(<TestComponent />);

    const button = screen.getByTestId('button');

    // The button should not have focus at start
    expect(button).not.toHaveFocus();

    // Open the drawer
    await user.click(button);

    // The input should have focus
    expect(screen.getByTestId('input')).toHaveFocus();

    // Close the drawer
    await user.click(screen.getByTestId('drawer-close-button'));

    // Wait for the button to be focused
    await waitFor(() => {
      expect(button).toHaveFocus();
    });
  });
});
