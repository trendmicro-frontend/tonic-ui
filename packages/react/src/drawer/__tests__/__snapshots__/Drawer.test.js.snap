// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render correctly 1`] = `
.emotion-0 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: var(--tonic-zIndices-drawer);
  top: 0;
  right: 0;
  bottom: 0;
}

.emotion-2 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, .7);
  opacity: 1;
  -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  outline: 0;
  overflow: clip;
  position: relative;
  color: var(--tonic-colors-white-primary);
  background: var(--tonic-colors-gray-90);
  border-width: var(--tonic-sizes-1q);
  border-style: solid;
  border-color: var(--tonic-colors-gray-80);
  box-shadow: 0 8px 32px 0 rgba(0, 0, 0, 0.48),0 4px 8px 0 rgba(0, 0, 0, 0.16);
  margin: auto;
  margin-right: 0;
  width: auto;
  height: 100%;
  max-width: 100vw;
  max-height: 100vh;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1);
  transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1);
}

.emotion-6 {
  padding-top: var(--tonic-space-4x);
  padding-bottom: var(--tonic-space-6x);
  padding-left: var(--tonic-space-4x);
  padding-right: var(--tonic-space-4x);
  position: relative;
  font-size: var(--tonic-fontSizes-xl);
  line-height: var(--tonic-lineHeights-xl);
}

.emotion-8 {
  padding-left: var(--tonic-space-4x);
  padding-right: var(--tonic-space-4x);
  padding-bottom: var(--tonic-space-6x);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: auto;
  overflow-y: auto;
}

.emotion-8:first-of-type {
  margin-top: calc(1rem + 1.75rem + .75rem);
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: var(--tonic-space-4x);
  padding-right: var(--tonic-space-4x);
  padding-top: var(--tonic-space-4x);
  padding-bottom: var(--tonic-space-4x);
}

.emotion-10:first-of-type {
  margin-top: calc(1rem + 1.75rem + .75rem);
}

.emotion-12 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: var(--tonic-borders-1);
  color: rgba(255, 255, 255, .47);
  cursor: pointer;
  outline: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-color: var(--tonic-colors-transparent);
  width: var(--tonic-sizes-8x);
  height: var(--tonic-sizes-8x);
  -webkit-transition: border-color 200ms cubic-bezier(0.4, 0, 0.2, 1),box-shadow 200ms cubic-bezier(0.4, 0, 0.2, 1),color 200ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: border-color 200ms cubic-bezier(0.4, 0, 0.2, 1),box-shadow 200ms cubic-bezier(0.4, 0, 0.2, 1),color 200ms cubic-bezier(0.4, 0, 0.2, 1);
  position: absolute;
  top: calc(.5rem - .0625rem);
  right: calc(.5rem - .0625rem);
}

.emotion-12:focus,
.emotion-12[data-focus] {
  border-color: #1e5ede;
  box-shadow: inset 0 0 0 1px #1e5ede;
  color: rgba(255, 255, 255, .47);
}

.emotion-12:hover,
.emotion-12[data-hover] {
  color: rgba(255, 255, 255, 1.0);
}

.emotion-12:focus:hover {
  color: rgba(255, 255, 255, 1.0);
}

.emotion-12:active,
.emotion-12[data-active] {
  color: rgba(255, 255, 255, .47);
}

.emotion-12:focus:active {
  border-color: #1e5ede;
  box-shadow: inset 0 0 0 1px #1e5ede;
  color: rgba(255, 255, 255, .47);
}

.emotion-14 {
  width: var(--tonic-sizes-4x);
  height: var(--tonic-sizes-4x);
  fill: currentColor;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-14:not(:root) {
  overflow: hidden;
}

.emotion-16 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  bottom: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-18 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  bottom: 0;
  right: 0;
}

.emotion-20 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  bottom: 0;
  left: 0;
}

.emotion-22 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.emotion-24 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  top: 0;
  left: 0;
}

.emotion-26 {
  box-sizing: border-box;
  max-height: 100%;
  max-width: 100%;
  overflow: hidden;
  position: fixed;
  z-index: var(--tonic-zIndices-toast);
  top: 0;
  right: 0;
}

<body>
  <div />
  <div
    class="tonic-ui-portal"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
    <div
      data-focus-lock-disabled="disabled"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <div
          class="emotion-2 emotion-1"
        />
        <div
          aria-modal="true"
          class="emotion-4 emotion-1"
          role="dialog"
          tabindex="-1"
        >
          <div
            class="emotion-6 emotion-1"
          >
            Drawer header
          </div>
          <div
            class="emotion-8 emotion-1"
          >
            Drawer body
          </div>
          <div
            class="emotion-10 emotion-1"
          >
            Drawer footer
          </div>
          <button
            aria-label="Close"
            class="emotion-12 emotion-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="emotion-14 emotion-15"
              focusable="false"
              role="presentation"
              viewBox="0 0 16 16"
            >
              <g>
                <path
                  d="M15 2l-1-1-6 6-6-6-0.99 0.99 5.99 6.010-6 6 1 1 6-6 6 6 1-1-6-6 6-6z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="-1"
    />
  </div>
  <div
    class="tonic-ui-portal"
  >
    <div
      class="emotion-16 emotion-1"
      data-toast-placement="bottom"
    />
    <div
      class="emotion-18 emotion-1"
      data-toast-placement="bottom-right"
    />
    <div
      class="emotion-20 emotion-1"
      data-toast-placement="bottom-left"
    />
    <div
      class="emotion-22 emotion-1"
      data-toast-placement="top"
    />
    <div
      class="emotion-24 emotion-1"
      data-toast-placement="top-left"
    />
    <div
      class="emotion-26 emotion-1"
      data-toast-placement="top-right"
    />
  </div>
</body>
`;
